File: CS50/week_1/mario-more/mario.c
#include <stdio.h>

void printPyramid(int height)
{
    int nSpaces = height - 1;
    for (int i = 0; i < height; i++)
    {
        for (int j = 0; j < nSpaces; j++)
        {
            printf(" ");
        }
        nSpaces--;
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }
        printf("  ");
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }
        printf("\n");
    }
}

int main(void)
{
    int height;
    do
    {
        printf("Height:\n>> ");
        scanf("%i", &height);
    }
    while (height < 1);
    printPyramid(height);
}

File: CS50/week_6/sentimental-mario-more/mario.py
def print_pyramid(size: int) -> None:
    for i in range(1, size + 1):
        val: str = "#" * i
        print(f"{val:>{size}}  {val}")


def main() -> None:
    while True:
        height: str = input("Height: ")
        if height.isdigit() and 9 > int(height) > 0:
            size: int = int(height)
            break
    print_pyramid(size)


if __name__ == "__main__":
    main()
